amalgamation/README.md

```diff
@@ -24,48 +24,114 @@ This module is created by [Jack Deng](https://github.com/jdeng).
 
 Android
 ---------------
-Setup NDK and build your standalone toolchain. [Instructions](http://developer.android.com/ndk/guides/standalone_toolchain.html#itc) Use the Advanced Method!!!
+Setup NDK and build your standalone toolchain. [Instructions](http://developer.android.com/ndk/guides/standalone_toolchain.html#itc) Use the Advanced Method!!! In particular set PATH, CC and CXX.
+
+Example:
+```
+export PATH=/tmp/my-android-toolchain/bin:$PATH
+export CC=arm-linux-androideabi-gcc   # or export CC=arm-linux-androideabi-clang
+export CXX=arm-linux-androideabi-g++  # or export CXX=arm-linux-androideabi-clang++
+```
+
 Build OpenBlas for Android: [Build OpenBlas](https://github.com/xianyi/OpenBLAS/wiki/How-to-build-OpenBLAS-for-Android)
+Modify OPENBLAS_ROOT in Makefile
 Type ```make ANDROID=1```
-Build will FAIL the first time as expected. Before this issue gets fixed, you need to manually remove include statements from mxnext_predict-all.cc and add <common.h> back. Here is the diff you want to apply:
```

Comment:
Yes, now you can run make just one time and it ends without errors. I have changed amalgation.py to automatically exclude the header files causing the errors.
Anyway the actual readme and the diff included are outdated, now you have to remove much fewer headers.